File: dongyimaimangerweb/src/main/webapp/js/controller/brandController.js
app.controller('brandController' ,function($scope,$http,brandService,$controller){

    $controller('baseController',{$scope:$scope});//继承

    //删除
    $scope.delete=function(){
        brandService.delete($scope.selectIds).success(
            function (response) {
                if (response.success){
                    $scope.reloadList();
                }else {
                    alert(response.message);
                }
            }
        )
    }

    //根据主键查询
    $scope.findOne=function(id){
        brandService.findOne(id).success(
            function (response) {
                $scope.entity=response;
            }
        );
    }

    //保存
    $scope.save = function(){

        var methodName="save";

        if ($scope.entity.id!=null){
            methodName="update";
        }

        brandService.save(methodName,$scope.entity).success(
            function (response) {
                if (response.success){
                    $scope.reloadList();
                }else {
                    alert(respones.message);
                }
            }
        )
    }

    // 查询所有
    $scope.findAll=function(){
        $http.get('../brand/findAll.do').success(
            function(response){
                $scope.list=response;
            }
        );
    }
    //分页查询
    $scope.findPages = function (pageNum,pageSize) {
        $http.get("../brand/findPages.do?pageNum="+pageNum+"&pageSize="+pageSize).success(
            function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems = response.total;
            }
        );
    }
    //模糊查询分页
    $scope.searchEntity = {};

    $scope.search = function(pageNum,pageSize){
        brandService.search(pageNum,pageSize,$scope.searchEntity).success(
            function (response) {
                $scope.list = response.rows;
                $scope.paginationConf.totalItems = response.total;
            }
        )
    }
});

File: dongyimaimangerweb/src/main/java/com/offcn/sellergoods/controller/BrandController.java
package com.offcn.sellergoods.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.offcn.entity.PageResult;
import com.offcn.entity.Result;
import com.offcn.pojo.TbBrand;
import com.offcn.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){

        return brandService.findAll();
    }
    @RequestMapping("/findPages")
    public PageResult findPages(int pageNum, int pageSize){

        return brandService.findPages(pageNum,pageSize);

    }
    @RequestMapping("/save")
    public Result save(@RequestBody TbBrand brand){

        try {
            brandService.save(brand);
            return new Result(true,"保存成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"保存失败");
        }
    }
    @RequestMapping("/findOne")
    public TbBrand findOne(long id){

        return brandService.findOne(id);
    }
    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){

        try {
            brandService.update(brand);

            return new Result(true,"更新成功");
        } catch (Exception e) {
            e.printStackTrace();

            return new Result(false,"更新失败");
        }
    }

    @RequestMapping("/delete")
    public Result delete(long[] ids){

        try {
            brandService.delete(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }

    @RequestMapping("/search")
    public PageResult search(@RequestBody TbBrand brand,int pageNum,int pageSize){
        return brandService.search(brand,pageNum,pageSize);
    }

    @RequestMapping("/selectOptionList")
    public List<TbBrand> selectOptionList(){
        return brandService.selectOptionList();
    }
}
